### PR TITLE
gnuapl: 1.6 -> 1.7

### PR DESCRIPTION
**Pull request by lsix**
See http://lists.gnu.org/archive/html/info-gnu/2017-03/msg00007.html
for release announcement.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by ndowens — CHANGES_REQUESTED** [review]
Remove with stdenv.lib; from meta and move to top then you can then remove stdenv.libbfrom optionals and optionalString

**Comment by vbgl** [comment]
@ndowens What is the rationale for such a refactoring? As it stands, this update looks good to me.

**Comment by ndowens** [comment]
It is just cosmetic thing I like to do, mostly just to use stdenv.lib once vs several times

**Comment by lsix** [comment]
I have no problem doing the refactoring. I’ll to that tonight.

**Comment by mdaiter** [comment]
@lsix @ndowens should we leave refactoring to another PR?

**Comment by ndowens** [comment]
I don't know why wait to do another PR just for cosmetic change?

**Comment by mdaiter** [comment]
@ndowens it primarily makes bookkeeping easier.

**Comment by ndowens** [comment]
Idk. If you change it and commit with msg cosmetic change or so that is bookkeeping IMO and can be split to fix something if it had to be

File: pkgs/development/interpreters/gnu-apl/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
-  version = "1.6";
+  version = "1.7";
 
   src = fetchurl {
     url = "mirror://gnu/apl/apl-${version}.tar.gz";
-    sha256 = "057zwzvvgcrrwsl52a27w86hgy31jqq6avqq629xj7yq90qah3ay";
+    sha256 = "07xq8ddlmz8psvsmwr23gar108ri0lwmw0n6kpxcv8ypas1f5xlg";
   };
 
   buildInputs = [ readline gettext ncurses ];
```
Comment (Mic92) [inline pkgs/development/interpreters/gnu-apl/default.nix]: "--disable-dependency-tracking" is added by default to configure.

Comment (lsix) [inline pkgs/development/interpreters/gnu-apl/default.nix]: I also do not see strong reason to keep the `"--disable-silent-rules"` (darwin only, but default behavior of automake). I’m going to drop it all together.

**Review by fpletz — APPROVED** [review]

**Pull request merged**